File: common/src/main/java/com/cc/dc/common/listener/HttpCallBack.java
package com.cc.dc.common.listener;

import io.reactivex.disposables.Disposable;

/**
 * Created by dc on 2017/9/21.
 * 网络请求完成的回调
 */
public interface HttpCallBack<T> {

    /**
     * 开始处理
     * @param disposable
     */
    void onStart(Disposable disposable);

    /**
     * 请求成功
     */
    void onResult(T result);

    /**
     * 请求失败
     */
    void onError();
}

File: kotlindemo/src/main/java/com/cc/dc/kotlindemo/net/HttpCallBack.kt
package com.cc.dc.kotlindemo.net

import io.reactivex.disposables.Disposable

/**
 * Created by dc on 2017/7/10.
 * 网络请求完成的回调
 */

interface HttpCallBack<T> {

    /**
     * 开始处理
     *
     * @param disposable
     */
    fun onStart(disposable: Disposable)

    /**
     * 请求成功
     */
    fun onResult(result: T?)

    /**
     * 请求失败
     */
    fun onError(msg: String)
}
